File: telerik_ui_for_winforms_courseware_chm/Courseware/Treeview/VB/DatabindingTreeView/Databinding/LoadOnDemand.vb
Imports Telerik.WinControls.Enumerations
Imports Telerik.WinControls.UI

Public Class LoadOnDemand
    Private Sub radTreeView1_NodesNeeded(ByVal sender As Object, ByVal e As Telerik.WinControls.UI.NodesNeededEventArgs) Handles radTreeView1.NodesNeeded
        ' measure how long this takes
        Dim stopwatch As New Stopwatch()
        stopwatch.Start()

        Dim count As Integer = 0

        ' get the data for all child nodes
        Dim weatherStations As String() = My.Resources.WeatherStations.Split(New String() {Constants.vbCrLf}, StringSplitOptions.RemoveEmptyEntries)


        For Each station As String In weatherStations
            e.Nodes.Add(New RadTreeNode(station))
            count += 1
        Next

        ' show performance results
        stopwatch.[Stop]()
        Dim ts As TimeSpan = stopwatch.Elapsed
        lblStatus.Text = [String].Format("{0:00}:{1:00}:{2:00}.{3:00} loading {4} nodes", ts.Hours, ts.Minutes, ts.Seconds, ts.Milliseconds / 10, count)

        Me.radTreeView1.EndUpdate()
    End Sub
End Class

File: telerik_ui_for_winforms_courseware_chm/Courseware/Treeview/CS/DatabindingTreeView/Databinding/LoadOnDemand.cs
using System;
using System.Diagnostics;
using Telerik.WinControls.Enumerations;
using Telerik.WinControls.UI;
using System.Collections.Generic;


namespace Databinding
{
    public partial class LoadOnDemand : RadForm
    {
        public LoadOnDemand()
        {
            InitializeComponent();
        }

        void radTreeView1_NodesNeeded(object sender, NodesNeededEventArgs e)
        {
            // measure how long this takes
            Stopwatch stopwatch = new Stopwatch();
            stopwatch.Start();

            int count = 0;

                // get the data for all child nodes
                string[] weatherStations =
                  Properties.Resources.WeatherStations.Split(
                  new string[] { "\r\n" }, StringSplitOptions.RemoveEmptyEntries);

                foreach (string station in weatherStations)
                {
                    e.Nodes.Add(new RadTreeNode(station));
                    count++;
                }
            
            // show performance results
            stopwatch.Stop();
            TimeSpan ts = stopwatch.Elapsed;
            lblStatus.Text = String.Format("{0:00}:{1:00}:{2:00}.{3:00} loading {4} nodes",
                ts.Hours, ts.Minutes, ts.Seconds, ts.Milliseconds / 10, count);
        }
    }
}
